uninstall

```diff
@@ -0,0 +1,214 @@
+end.parse!
+
+if $HOMEBREW_PREFIX_CANDIDATES.empty? # Attempt to locate Homebrew unless `--path` is passed
+  p = `brew --prefix` rescue ""
+  $HOMEBREW_PREFIX_CANDIDATES << Pathname.new(p.strip) unless p.empty?
+  p = `command -v brew`
+  $HOMEBREW_PREFIX_CANDIDATES << Pathname.new(p.strip).parent.parent unless p.empty?
+  $HOMEBREW_PREFIX_CANDIDATES << Pathname.new("/usr/local") # Homebrew default path
+  $HOMEBREW_PREFIX_CANDIDATES << Pathname.new("#{ENV["HOME"]}/.linuxbrew") # Linuxbrew default path
+end
+
+$HOMEBREW_PREFIX = $HOMEBREW_PREFIX_CANDIDATES.detect do |p|
+  p.directory? && (p/"Library/Homebrew").directory?
+end
+
+abort "Failed to locate Homebrew" if $HOMEBREW_PREFIX.nil?
+
+$HOMEBREW_FILES = %W[
+  .git .gitignore .yardopts CODEOFCONDUCT.md CONTRIBUTING.md
+  LICENSE.txt README.md SUPPORTERS.md Cellar Library/.rubocop.yml
+  Library/Aliases Library/Contributions Library/ENV Library/Formula
+  Library/Homebrew Library/LinkedKegs Library/Taps Library/brew.rb
+  bin/brew share/doc/homebrew share/man/man1/brew.1
+].map { |p| $HOMEBREW_PREFIX/p }
+
+unless options[:skip_cache_log]
+  $HOMEBREW_FILES += %W[
+    #{ENV["HOME"]}/Library/Caches/Homebrew
+    #{ENV["HOME"]}/Library/Logs/Homebrew
+    /Library/Caches/Homebrew
+    #{ENV["HOME"]}/.cache/Homebrew
+  ].map { |p| Pathname.new(p) }
+end
+
+if /darwin/i === RUBY_PLATFORM
+  $HOMEBREW_FILES += %W[
+    /Applications
+    #{ENV["HOME"]}/Applications
+  ].map do |p|
+    Pathname.new(p).subdirs.select do |app|
+      app.resolved_path.to_s.start_with? $HOMEBREW_PREFIX.to_s
+    end
+  end.flatten
+end
+
+$HOMEBREW_FILES.select!(&:exist?)
+
+unless options[:quiet]
+  warn "This script #{options[:dry_run] ? "would" : "will"} remove:"
+  $HOMEBREW_FILES.each(&:pretty_print)
+end
+
+unless options[:force] || options[:dry_run]
+  STDERR.print "Confirm to uninstall Homebrew? [y/N] "
+  abort unless gets.rstrip =~ /y|yes/i
+end
+
+ohai "Unlink Homebrew Installation..." unless options[:quiet]
+paths = %W[Frameworks bin etc include lib opt sbin share var].
+  map { |p| $HOMEBREW_PREFIX/p }.select(&:exist?).map(&:to_s)
+args = %W[-E] + paths + %W[-regex .*/info/([^.][^/]*\.info|dir)]
+if options[:dry_run]
+  args << "-print"
+else
+  args += %W[-exec /bin/bash -c]
+  args << "/usr/bin/install-info --delete --quiet {} \"$(dirname {})/dir\""
+  args << ";"
+end
+puts "Would uninfo:" if options[:dry_run]
+system "/usr/bin/find", *args
+args = paths + %W[-type l -lname */Cellar/*]
+if options[:dry_run]
+  args << "-print"
+else
+  args += %W[-exec unlink {} ;]
+end
+puts "Would unlink:" if options[:dry_run]
+system "/usr/bin/find", *args
+
+ohai "Remove Homebrew Installation..." unless options[:quiet]
+$HOMEBREW_FILES.each do |file|
+  if options[:dry_run]
+    puts "Would remove #{file}"
+    next
+  end
+
+  if file.symlink?
+    system "unlink", file.to_s
+  elsif file.directory?
+    system "rm", "-rf", file.to_s
+  else
+    system "rm", file.to_s
+  end
```

Comment:
`system "unlink"` is the same as `system "rm"` (except unlink doesn't accept flags and only accepts one file argument).

But I think all three branches can be removed in favor of just calling `FileUtils.rm_rf(file)` or equivalent.